src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -136,6 +138,9 @@ export class Kernel implements IKernel {
     public async executeCell(cell: NotebookCell): Promise<void> {
         const stopWatch = new StopWatch();
         const notebookPromise = this.startNotebook({ disableUI: false, document: cell.notebook });
+        if (cell.notebook.notebookType === InteractiveWindowView) {
```

Comment:
This is a very weird pr, no part of the code makes me think that any of these changes will update the execution count 
My only guess is that when we add a cell using the cell hash provided it's a real Block of code, else it's treated as silent and we end up with no execution count.

Is that right?
If that's the case then cell hash provided does a whole lot more than it seems to.
No need to change this, but I would have never guessed, and to me is a pretty weird way to wire things up.

Reply:
Yeah the executionCount is state that's maintained inside the cellHashProvider. I agree that it's pretty opaque. FWIW, this used to be done in an INotebookExecutionLogger, I think my change to explicitly call addCellHash made it more explicit that the cellHashProvider is getting called, so the code now scans funny.